dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.portlets.languagesmanager.model.DisplayedLanguage;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Constants;
+import com.dotmarketing.util.InodeUtils;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.PortletURLUtil;
+import com.dotmarketing.util.StringUtils;
+import com.dotmarketing.util.UtilMethods;
+import com.dotmarketing.util.WebKeys;
+
+import java.io.File;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+import org.apache.velocity.context.Context;
+import org.apache.velocity.exception.ParseErrorException;
+import org.apache.velocity.exception.ResourceNotFoundException;
+import org.apache.velocity.tools.view.ToolboxManager;
+import org.apache.velocity.tools.view.context.ChainedContext;
+import org.apache.velocity.tools.view.servlet.ServletToolboxManager;
+
+import com.liferay.portal.model.User;
+import com.liferay.util.SystemProperties;
+
+
+public class VelocityUtil {
+
+	private static VelocityEngine ve = null;
+	private static boolean DEFAULT_PAGE_TO_DEFAULT_LANGUAGE = LanguageWebAPI.canDefaultPageToDefaultLanguage();
+	
+	private synchronized static void init(){
+		if(ve != null)
+			return;
+		ve = new VelocityEngine();
+		try{
+			ve.init(SystemProperties.getProperties());
+
+			Logger.debug(VelocityUtil.class, SystemProperties.getProperties().toString());
+		}catch (Exception e) {
+			Logger.error(VelocityUtil.class,e.getMessage(),e);
+		}
+	}
+	
+	public static VelocityEngine getEngine(){
+		if(ve == null){
+			init();
+			if(ve == null){
+				Logger.fatal(VelocityUtil.class,"Velocity Engine unable to initialize : THIS SHOULD NEVER HAPPEN");
+				throw new DotRuntimeException("Velocity Engine unable to initialize : THIS SHOULD NEVER HAPPEN");
+			}
+		}
+		return ve;
+	}
+	/**
+	 * Changes $ and # to velocity escapes.  This is helps filter out velocity code injections.
+	 * @param s 
+	 * @return
+	 */
+	public static String cleanVelocity(String s) {
+		if (s==null) {
+			return null;
+		}
+		s=s.replace("$", "${esc.dollar}");
+		s=s.replace("#", "${esc.hash}");
+		return s;
+	}
+
+
+	
+	public String parseVelocity(String velocityCode, Context ctx){
+		VelocityEngine ve = VelocityUtil.getEngine();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "ve" which hides the field declared at line 60. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)